packages/autorest.python/autorest/codegen/templates/model_base.py.jinja2

```diff
@@ -715,6 +705,17 @@ def _get_deserialize_callable_from_annotation(  # pylint: disable=R0911, R0915,
 
         return functools.partial(_deserialize_with_union, deserializers)
 
+    # is it optional?
+    try:
+        if any(a for a in annotation.__args__ if a == type(None)):  # pyright: ignore
+            if_obj_deserializer = _get_deserialize_callable_from_annotation(
+                next(a for a in annotation.__args__ if a != type(None)), module, rf  # pyright: ignore
+            )
+
+            return functools.partial(_deserialize_with_optional, if_obj_deserializer)
+    except AttributeError:
+        pass
+
```

Comment:
imo, i prefer to copy the sort logic from union to optional to solve the enum problem. plus, we need to put all basic type to the end i think, not only `string`, but `int`, `float`, etc. plus, you need to add another `None` test for this optional property.

Reply:
@tadelesh  Make sense and updated